utils_cv/action_recognition/model.py

```diff
@@ -229,28 +233,22 @@ def fit(self, train_cfgs):
             )
 
         # Learning rate scheduler
-        scheduler = None
-        warmup_pct = train_cfgs.get('warmup_pct', None)
-        lr_decay_steps = train_cfgs.get('lr_decay_steps', None)
-        if warmup_pct is not None:
+        if train_cfgs.get('use_one_cycle_policy', False):
             # Use warmup with the one-cycle policy
-            lr_decay_total_steps = train_cfgs.epochs if lr_decay_steps is None else lr_decay_steps
             scheduler = torch.optim.lr_scheduler.OneCycleLR(
                 optimizer,
                 max_lr=train_cfgs.lr,
-                total_steps=lr_decay_total_steps,
+                total_steps=train_cfgs.epochs,
                 pct_start=train_cfgs.get('warmup_pct', 0.3),
-                base_momentum=0.9*train_cfgs.momentum,
-                max_momentum=train_cfgs.momentum,
-                final_div_factor=1/train_cfgs.get('lr_decay_factor', 0.0001),
+                base_momentum=0.9*momentum,
+                max_momentum=momentum,
             )
-        elif lr_decay_steps is not None:
-            lr_decay_total_steps = train_cfgs.epochs
+        else:
             # Simple step-decay
             scheduler = torch.optim.lr_scheduler.StepLR(
                 optimizer,
-                step_size=lr_decay_steps,
-                gamma=train_cfgs.get('lr_decay_factor', 0.1),
+                step_size=train_cfgs.get('lr_step_size', None),
```

Comment:
If `lr_step_size` doesn't exist as a config parameter, it will cause failures down the road when retrieving the learning rate.

Is there a reason why we aren't falling back to a default learning rate if the config doesn't exist?

Reply:
Thanks for finding. None is not a valid input to step_size, so that'd be throwing an error if train_cfgs.lr_step_size isn't defined.
This parameter will reduce the learning rate by a factor of e.g. 10x, after a certain number of steps. Will edit the code so that this learning rate reduction is deactivated if train_cfgs.lr_step_size isn't provided